app/org/sagebionetworks/bridge/play/controllers/ParticipantController.java

```diff
@@ -100,8 +98,14 @@ public Result getParticipant(String userId) {
         UserSession session = getAuthenticatedSession(RESEARCHER);
         Study study = studyService.getStudy(session.getStudyIdentifier());
         
-        StudyParticipant participant = participantService.getParticipant(study,
-                session.getParticipant().getRoles(), userId);
+        StudyParticipant participant = participantService.getParticipant(study, userId, true);
+        
+        // if not enabled, do not include the health code in data returned from this API. 
+        if (!study.isHealthCodeExportEnabled()) {
+            participant = new StudyParticipant.Builder().copyOf(participant)
+                    .withHealthCode(null).build();
+        }
+        
```

Comment:
For now I'm removing healthCode and encryptedHealthCode from the JSON just to be sure this happens. This can be replaced with a Jackson filter in a follow-on update.